### DISCUSSION TITLE
Add a hardware version to the device registry

### DISCUSSION THREAD

**Discussion by bdraco** [home-assistant/architecture#685] [Entity Models] [answered]
### Context

Since some devices behave differently between hardware versions this is often critical in determining the root cause of an issue.  

Finding the hardware revision can be difficult as manufacturers tend not to make it obvious as they have a financial incentive to be vague about it given they don’t want people to not buy the old stock which means they are sometimes hidden a few menus deep or only available via API. 

This usually means the users have to do something like:
Manually an API call
Dump debug logs and find the version
Broadcast some type of magic packet
Dig through menus

### Proposal
Add a `hw_version` field to the device registry to store the hardware version of a device, and display it in the UI so it is easy for users to report an issue. see patterns, or understand why their setup doesn’t work for one version of a device but does work for another version.

Backend implementation:  https://github.com/home-assistant/core/pull/61650
Dev docs: home-assistant/developers.home-assistant#1156
Frontend: home-assistant/frontend#10914

### Consequences
We can see patterns a bit easier between hardware versions.
There is a small but imperceptible performance impact with additional data written to disk.

**Comment by Jc2k** [comment]
+1

In homekit there is "hardware revision". It's a string. I think it's fairly free form about what is in there, so it would be super easy to add this to homekit_controller.

Would you expose it in homekit bridge too?

**Reply by bdraco** [reply]
> Would you expose it in homekit bridge too?

Yes that would be part of the goal.

**Comment by emontnemery** [comment, marked as answer]
This is a good proposal, I suggest to approve it as is.
